equals and hashCode methods added to Rectangle class

MykolaKatsai opened the pull request.

MykolaKatsai committed:
equals and hashCode methods added to Rectangle class

boroda4436 commented on src/main/java/core/basesyntax/Rectangle.java:
```diff
@@ -27,4 +27,16 @@ public Integer getLength() {
         return color;
     }
+
+    public boolean equals(Rectangle rec) {
```
@NikolayKatsai what if rec == null?
Is this method overrided from Object?

MykolaKatsai committed:
equals and hashCode methods added to Rectangle class

MykolaKatsai commented:
Доброго вечора, чомусь при запуску через мавен, при виклику в тесті методу equals, вилітає ексепшн:

java.lang.NoSuchMethodError: core.basesyntax.Rectangle.equals(Lcore/basesyntax/Rectangle;)Z
        at core.basesyntax.RectangleTest.equalityOfRectangle(RectangleTest.java:23


оскільки це я ніяк не міг пояснити вирішив перевірити тут, і дійсно схоже помилка у мавені

MykolaKatsai commented:
а якщо написати equals(Rectangl r)  замість equals(Object o) то ексепшену не буде

boroda4436 commented on src/main/java/core/basesyntax/Rectangle.java:
```diff
@@ -27,4 +27,22 @@ public Integer getLength() {
+        }
+        if (obj instanceof Rectangle) {
+            Rectangle rec = (Rectangle) obj;
+            if (width.equals(rec.getWidth()) && length.equals(rec.getLength())
```
@NikolayKatsai Simplify this if.

boroda4436 commented on src/main/java/core/basesyntax/Rectangle.java:
```diff
@@ -27,4 +27,22 @@ public Integer getLength() {
+            return true;
+        }
+        if (obj instanceof Rectangle) {
+            Rectangle rec = (Rectangle) obj;
```
@NikolayKatsai Not useful variable name

boroda4436 commented on src/main/java/core/basesyntax/Rectangle.java:
```diff
@@ -27,4 +27,22 @@ public Integer getLength() {
+    }
+
+    public int hashCode() {
+        return width * length + width / length + color.hashCode();
```
@NikolayKatsai Read about prime numbers in hashcode

MykolaKatsai committed:
equals method changed in Rectangle class:
variable rec changed on comparedRectangle;

hashCode method changed in Rectangle class

MykolaKatsai committed:
checking in equals method in Rectangle class simplified
